src/wp-admin/includes/image.php

```diff
@@ -261,8 +261,10 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		$image_meta['image_meta'] = $exif_meta;
 	}
 
+	$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+
 	// Do not scale (large) PNG images. May result in sub-sizes that have greater file size than the original. See #48736.
-	if ( 'image/png' !== $imagesize['mime'] ) {
+	if ( 'image/png' !== $imagesize['mime'] || 'image/png' !== $output_format ) {
```

Comment:
Shouldn't this use `&&` instead? Also, `$output_format` can be an array.

Reply:
this now does the resizing of the full size image in the uploaded image mime is a PNG OR if the output type will be a  PNG.

So this is a `||` not '&&`. && would mean the resize only happened when both teh input and the output where non PNG.